### PR TITLE
storaged: enable modules with the non-deprecated udisks API

### PR DESCRIPTION
**Pull request by jelly**
Since 2.9.0 EnableModules has been deprecated and replaced by EnableModule which can only enable a given module.

---

It would be cool if the `pkg/storaged/$thing` enabled itself but that becomes a bit hairy with juggling client state around.

2.9.0 is packaged in RHEL 8 so should be available everywhere we need it.

### CODE REVIEW

File: pkg/storaged/client.js [hunk 1]
```diff
@@ -854,29 +854,42 @@ function init_model(callback) {
                 });
     }
 
-    function enable_udisks_features() {
+    async function enable_udisks_features() {
         if (!client.manager.valid)
             return Promise.resolve();
-        if (!client.manager.EnableModules)
+        if (!client.manager.EnableModule)
```
Comment (jelly) [inline pkg/storaged/client.js:860]: This check seems a bit strange, not sure what we guard against? Shall we just drop it?

File: pkg/storaged/client.js [hunk 2]
```diff
@@ -854,29 +854,42 @@ function init_model(callback) {
...
+
+        try {
+            await client.manager.EnableModule("btrfs", true);
+            client.manager_btrfs = proxy("Manager.BTRFS", "Manager");
+            await client.manager_btrfs.wait();
+            client.features.btrfs = client.manager_btrfs.valid;
+            if (client.features.btrfs)
+                btrfs_start_polling();
+        } catch (error) {
+            console.warn("Can't enable storaged btrfs module", error.toString());
```
Comment (cockpituous) [inline pkg/storaged/client.js:871]: These 2 added lines are not executed by any test.

File: pkg/storaged/client.js [hunk 3]
```diff
@@ -854,29 +854,42 @@ function init_model(callback) {
...
+        }
+
+        try {
+            await client.manager.EnableModule("iscsi", true);
+            client.manager_iscsi = proxy("Manager.ISCSI.Initiator", "Manager");
+            await client.manager_iscsi.wait();
+            client.features.iscsi = (client.manager_iscsi.valid && client.manager_iscsi.SessionsSupported !== false);
+        } catch (error) {
+            console.warn("Can't enable storaged iscsi module", error.toString());
```
Comment (cockpituous) [inline pkg/storaged/client.js:880]: These 2 added lines are not executed by any test.

File: pkg/storaged/client.js [hunk 4]
```diff
@@ -854,29 +854,42 @@ function init_model(callback) {
...
+        }
+
+        try {
+            await client.manager.EnableModule("lvm2", true);
+            client.manager_lvm2 = proxy("Manager.LVM2", "Manager");
+            await client.manager_lvm2.wait();
+            client.features.lvm2 = client.manager_lvm2.valid;
+        } catch (error) {
+            console.warn("Can't enable storaged lvm2 module", error.toString());
```
Comment (cockpituous) [inline pkg/storaged/client.js:889]: These 2 added lines are not executed by any test.

File: pkg/storaged/client.js [hunk 5]
```diff
@@ -854,29 +854,42 @@ function init_model(callback) {
...
+        }
+
+        return Promise.resolve();
```
Comment (mvollmer) [inline pkg/storaged/client.js:892]: See above, no Promises returned from `async` functions.

File: pkg/storaged/client.js (hunk 1 above)
Comment (mvollmer) [inline pkg/storaged/client.js:860]: Let's remove this check. We do have 2.9 everywhere, right? :-)

File: pkg/storaged/client.js [hunk 6]
```diff
@@ -854,29 +854,38 @@ function init_model(callback) {
...
+
+        try {
+            await client.manager.EnableModule("btrfs", true);
+            client.manager_btrfs = proxy("Manager.BTRFS", "Manager");
+            await client.manager_btrfs.wait();
+            client.features.btrfs = client.manager_btrfs.valid;
+            if (client.features.btrfs)
+                btrfs_start_polling();
+        } catch (error) {
+            console.warn("Can't enable storaged btrfs module", error.toString());
```
Comment (cockpituous) [inline pkg/storaged/client.js:869]: These 2 added lines are not executed by any test.

File: pkg/storaged/client.js [hunk 7]
```diff
@@ -854,29 +854,38 @@ function init_model(callback) {
...
+        }
+
+        try {
+            await client.manager.EnableModule("iscsi", true);
+            client.manager_iscsi = proxy("Manager.ISCSI.Initiator", "Manager");
+            await client.manager_iscsi.wait();
+            client.features.iscsi = (client.manager_iscsi.valid && client.manager_iscsi.SessionsSupported !== false);
+        } catch (error) {
+            console.warn("Can't enable storaged iscsi module", error.toString());
```
Comment (cockpituous) [inline pkg/storaged/client.js:878]: These 2 added lines are not executed by any test.

File: pkg/storaged/client.js [hunk 8]
```diff
@@ -854,29 +854,38 @@ function init_model(callback) {
...
+        }
+
+        try {
+            await client.manager.EnableModule("lvm2", true);
+            client.manager_lvm2 = proxy("Manager.LVM2", "Manager");
+            await client.manager_lvm2.wait();
+            client.features.lvm2 = client.manager_lvm2.valid;
+        } catch (error) {
+            console.warn("Can't enable storaged lvm2 module", error.toString());
```
Comment (cockpituous) [inline pkg/storaged/client.js:887]: These 2 added lines are not executed by any test.

**Review by mvollmer — APPROVED** [review]
Thanks!

**Pull request merged**